### DISCUSSION TITLE
Export all workspace content programmatically

### DISCUSSION THREAD

**Discussion by 5hay** [Devorein/Nishan#167] [Q&A] [answered]
Hi,

First of all, thank you for your efforts. Even though the official API is already in private beta, I don't have much faith they will release it anytime soon.

I need to do regular backups of my workspace. In the settings of Notion there is the button "Export all workspace content". I've looked through the Docs, but couldn't find if that is already possible with this project. 

If not, could you imagine implementing it anytime soon? Alternatively, I could just hack something together with puppeteer, but I wanted to ask here first. ✌

**Comment by 5hay** [comment, marked as answer]
This request is no longer relevant. I wrote a small tool in Go that can backup (part of) the workspace. If someone comes across this post and also has a need for something like that: https://github.com/5hay/notionbackup

**Reply by Devorein** [reply]
Hey extremely sorry for the late response. I was not aware of this discussion at all since the GitHub UI doesn't show little numbers besides discussions like they do for issues and such. I'm glad that you managed to do it by yourself. But if you ever need the code, here it is.

```ts
import { NotionTasks } from '@nishans/tasks';

const token ='token_v2';

async function main () {
	const export_url = await NotionTasks.Export.space(
		'space_id',
		{
			exportType: 'markdown',
			locale: 'en',
			timeZone: 'timezone',
		},
		{
			token,
			user_id: 'user_id',
			logger: true
		}
	);
	console.log(export_url);
}
main();
``` 

If you navigate to the `export_url` your browser will automatically download a zip containing all the workspace content. Again sorry for the trouble :(

**Reply by 5hay** [reply]
Hey,

thank you very much for the reply. Don't worry, I was already kinda thinking this would be the case 😄. I'm glad your project is also able to do that. I got a few more projects in mind, so this library is greatly appreciated. Have a nice weekend.
